kit/Kit.cpp

```diff
@@ -1285,6 +1309,170 @@ void Document::updateActivityHeader() const
+{
+    if (!joinThreads())
+    {
+        LOG_WRN("Failed to join threads before async save");
+        return false;
+    }
+
+    size_t threads = getCurrentThreadCount();
+    if (threads != 1)
+    {
+        LOG_WRN("Failed to ensure we have just one, we have: " << threads);
+        return false;
+    }
+
+#if 0
+    // TODO: compare FD count in a normal process with how
+    // many we see open now.
+    int expectFds = 2 // SocketPoll wakeups
+        + 1; // socket to coolwsd
+    int actualFds = fdCounter->count();
+    if (actualFds != expectFds)
+    {
+        LOG_WRN("Can't background save: " << actualFds << " fds open; expect " << expectFds);
+        return false;
+    }
+#endif
+
+    const auto start = std::chrono::steady_clock::now();
+
+    // TODO: close URPtoLoFDs and URPfromLoFDs and test
+    if (isURPEnabled())
+    {
+        LOG_WRN("Can't background save with URP enabled");
+        return false;
+    }
+
+    // FIXME: only do one of these at a time ...
+    // FIXME: defer and queue a 2nd save if queued during save ...
+
+    std::shared_ptr<StreamSocket> parentSocket, childSocket;
+    if (!StreamSocket::socketpair(parentSocket, childSocket))
+        return false;
+
+    // To encode into the child process id for debugging
+    static size_t numSaves = 0;
+    numSaves++;
+
+    const pid_t pid = fork();
+
+    if (!pid) // Child
+    {
+        // sort out thread local variables to get logging right from
+        // as early as possible.
+        Util::setThreadName("kitbgsv_" + Util::encodeId(_mobileAppDocId, 3) +
+                            "_" + Util::encodeId(numSaves, 3));
+        _isBgSaveProcess = true;
+
+        SigUtil::addActivity("forked background save process: " +
+                             std::to_string(pid));
+
+        childSocket.reset();
+        // now we just have a single socket to our parent
+
+        // Hard drop our previous connections to coolwsd and shared wakeups.x
+        KitSocketPoll::cleanupChildProcess();
+
+        // close duplicate kit->wsd socket
+        auto kitWs = std::static_pointer_cast<KitWebSocketHandler>(_websocketHandler);
+        kitWs->shutdownForBackgroundSave();
+
+        // now send messages to the parent instead of the kit.
+        auto parentWs = std::make_shared<BgSaveChildWebSocketHandler>("bgsv_child_ws");
+        parentSocket->setHandler(parentWs);
+        parentSocket->setWebSocket(); // avoid http upgrade.
+        _saveProcessParent = parentWs;
+
+        // hand parentSocket to the main poll
+        KitSocketPoll::getMainPoll()->insertNewSocket(parentSocket);
+        parentWs.reset();
+
+        getLOKit()->setForkedChild(true);
+
+        const auto now = std::chrono::steady_clock::now();
+        LOG_TRC("Background save process " << getpid() << " fork took " <<
+                std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() << "ms");
```

Comment:
Chrono serializes with the units already. Removing `.count() << "ms"` should do the right thing and log "ms" as expected).

Reply:
oh nice =) fixed.